src/Nethermind/Nethermind.Taiko/TaikoPayloadPreparationService.cs

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Consensus.Producers;
+using Nethermind.Core;
+using Nethermind.Core.Timers;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.BlockProduction;
+
+namespace Nethermind.Taiko;
+
+public class TaikoPayloadPreparationService(
+    PostMergeBlockProducer blockProducer,
+    ITimerFactory timerFactory,
+    ILogManager logManager,
+    TimeSpan timePerSlot,
+    IL1OriginStore l1OriginStore) : PayloadPreparationService(
+        blockProducer,
+        NoBlockImprovementContextFactory.Instance,
+        timerFactory,
+        logManager,
+        timePerSlot)
+{
+    private readonly ILogger _logger = logManager.GetClassLogger();
+    private readonly IL1OriginStore _l1OriginStore = l1OriginStore;
+
+    protected override void ImproveBlock(string payloadId, BlockHeader parentHeader,
```

Comment:
A bit confused here as Taiko were not to produce blocks? Would benefit from detailed comment explanation.